### ISSUE TITLE
name of List field gets truncated when mapping to record

### ISSUE THREAD

**Issue by agschaid** [mapstruct/mapstruct#2589]
Hi,

I have an issue where field names get truncated when:

- the field has an generic type (I tried List<>, Set<> and I am suspecting that custom types should fail as well)
- the target is a java record

To reproduce create ClassA and ClassB like this:

```
public class ClassA {

	private String someString;
	private List<String> stringList;

	public String getSomeString() { return someString; }
	public void setSomeString(String someString) { this.someString = someString; }
	public List<String> getStringList() { return stringList; }
	public void setStringList(List<String> stringList) { this.stringList = stringList; }

}
```
```
public record ClassB(
	String someString,
	List<String> stringList
) {};

```

Then create the following mapper:

```
@Mapper
public interface ABMapper {
  ClassA bToA(ClassB classB);
  ClassB aToB(ClassA classA);
}

```

You'll get a compiler error (pointing to the `aToB` function) that says: ` Unmapped target property: "ingList". `
Obviously the first 3 letters of "stringList" have been dropped when mapping into a record.

My mapstruct version is 1.4.2.Final

**Comment by lostiniceland** [comment]
I can confirm the issue. Seeing the exact same result.
It is not relatated to list fields but rather to the last record entry.

**Comment by KENNYSOFT** [comment]
May duplicate of #2554

**Comment by agschaid** [comment]
> May duplicate of #2554

appears to be the same issue. So this will probably fixed in 1.5

**Comment by lostiniceland** [comment]
@agschaid could you release a Beta2 for 1.5 on Maven Central with the fix from #2554 included. Beta1 is from July. Or is 1.5 due soon?

**Comment by agschaid** [comment]
@lostiniceland I am not a mapstruct developer. I just closed my own issue since it's a duplicate of an issue that is about to be closed.
